### PR TITLE
Character Counter

### PR DESCRIPTION
**Pull request by andycochran**
This PR adds a character counter component.

If the value approaches the limit it shows a warning:

image

If is it over the limit, that warning turns bright red: 

image

### CODE REVIEW

File: client/app/components/character-counter.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  get warningClass() {
+    if (this.stringLength > this.args.maxlength) {
+      return "invalid"
+    };
+    if ((this.stringLength / this.args.maxlength) > 0.9) {
+      return "warning"
+    };
+    return
```
Comment (allthesignals) [inline client/app/components/character-counter.js:15]: maybe just `return '';`? idk, lint will not like this

File: client/tests/integration/components/character-counter-test.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    assert.equal(this.element.textContent.trim(), '');
+
+    // Template block usage:
+    await render(hbs`
+      <CharacterCounter>
+        template block text
+      </CharacterCounter>
+    `);
+
+    assert.equal(this.element.textContent.trim(), 'template block text');
```
Comment (allthesignals) [inline client/tests/integration/components/character-counter-test.js:24]: aw, c'mon. it'll take 1 minute to add a test?

File: client/app/components/character-counter.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import Component from '@glimmer/component';
+
+export default class CharacterCounterComponent extends Component {
+  get stringLength() {
+    return (this.args.string ? this.args.string.length : 0);
```
Comment (allthesignals) [inline client/app/components/character-counter.js:5]: nice

File: client/app/components/character-counter.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  get warningClass() {
+    if (this.stringLength > this.args.maxlength) {
+      return "invalid"
+    };
+    if ((this.stringLength / this.args.maxlength) > 0.9) {
```
Comment (allthesignals) [inline client/app/components/character-counter.js:12]: cool

File: client/app/components/character-counter.js (hunk 1 above)
Comment (andycochran) [reply client/app/components/character-counter.js:15]: `return null`?

File: client/tests/integration/components/character-counter-test.js (hunk 2 above)
Comment (andycochran) [reply client/tests/integration/components/character-counter-test.js:24]: 😇

Comment (andycochran) [reply client/tests/integration/components/character-counter-test.js]: I added three. 😎

**Pull request merged**